BZ1873575: add a section called reinstalling the OCP cluster (#38587)
openshift/openshift-docs: modules/restarting-installation.adoc

// Module included in the following assemblies:
//
// installing/installing-troubleshooting.adoc

[id="restarting-installation_{context}"]
= Reinstalling the {product-title} cluster

> If you are unable to debug and resolve issues in the failed {product-title} installation, you may need to consider installing a new {product-title} cluster. Before starting the installation process again, you must do a thorough cleanup.

Reviewer: ```suggestion
If you are unable to debug and resolve issues in the failed {product-title} installation, you might need to consider installing a new {product-title} cluster. Before starting the installation process again, you must do a thorough cleanup.
```

Author: Comment addressed.

.Procedure

1. Destroy the cluster and remove all the resources associated with the cluster, including the hidden installer state files in the installation directory:
+
[source, terminal]
----

> $ ./openshift-install destroy cluster --dir=<installation_directory> <1>

Reviewer 2: This command is only applied on IPI cluster.  For UPI cluster, user need to cleanup the cluster and associated resources by themselves based on how they create the infrastructure.

----
<1> `installation_directory` is the directory you specified when you ran `./openshift-install create cluster`. This directory contains the {product-title}

> definition files that the installation program creates.

Reviewer 2: After running above comment, it's better to add this part "And if you want to reinstall from scratch, rm -rf the asset directory beforehand." suggested in dev doc https://github.com/openshift/installer/tree/ee89123890d894201145d0c5176a631885f08220#cleanup, especially for fresh UPI installation

> 2. xref:../installing/index.adoc#ocp-installation-overview[Install a new {product-title} cluster].

Reviewer: We still can't put xrefs in modules, unfortunately. Since these steps already exist, I would suggest putting "Follow procedure for Installing a new {product-title} cluster." as step 2. Then in the assembly file, add an additional resource section and write: For details, see xref:....[Install a new {product-title} cluster]. Since this is the last module on the page, the Additional Resources should be perfectly positioned under that step 2. Let me know if I can clarify anything.

Author: Thanks for the review. Addressed this comment.

// Module included in the following assemblies:
//
// installing/installing-troubleshooting.adoc

[id="restarting-installation_{context}"]
= Reinstalling the {product-title} cluster

> If you are unable to debug and resolve issues in the failed {product-title} installation, you might need to consider installing a new {product-title} cluster. Before starting the installation process again, you must do a thorough cleanup.

Reviewer 3: ```suggestion
If you are unable to debug and resolve issues in a failed {product-title} installation, consider installing a new {product-title} cluster. Before starting the installation process again, you must complete thorough cleanup.
```

Changes suggested for more minimalistic and direct writing.

Author: Addressed.

> For the user-provisioned infrastructure (UPI) installation, you must manually destroy the cluster and delete all the associated resources. For the installer-provisioned infrastructure (IPI) installation, see the following procedure.

Reviewer 3: ```suggestion
For an user-provisioned infrastructure (UPI) installation, you must manually destroy the cluster and delete all associated resources. The following procedure is for an installer-provisioned infrastructure (IPI) installation.
```

Author: Addressed.

.Procedure

> 1. Destroy the cluster and remove all the resources associated with the cluster, including the hidden installer state files in the installation directory:

Reviewer 3: ```suggestion
. Destroy the cluster and remove all the resources associated with the cluster, including the hidden installer state files in the installation directory:
```

Remove the hard-coded "1." and use "." instead. This applies to all steps below.

Author: Addressed.
